F-08 Выполнена

Feintty opened the pull request.

Feintty committed:
Merge pull request #3 from Feintty/review

V-07

Feintty committed:
F-08 Создано основное хранилище
F-08 Созданы редюсеры для авторизации
F-08 Созданы экшены для полей ввода

Feintty committed:
F-08 Реализована авторизация пользователя
F-08 Разработан snackbar для уведомления пользователей

Feintty committed:
F-08 Вынесены ключи в окружение

Feintty committed:
F-08 Реализована запись куки и проверка на их наличие
F-08 Реализовано взаимодействие snackbara по его id
F-08 Реализовано разграничение  по роутам
F-08 Небольшой рефакторинг

konkonskiy commented on src/App.tsx:
```diff
@@ -1,10 +1,33 @@
+          message="Пользователь не найден!"
+        />
+      );
+    } else if (isLogged) {
```
Ошибок может быть огромное количество. Представляешь какой длины будет условная конструкция? 
Попробуй отслеживать тип ошибки.

konkonskiy replied:
Не исправил

konkonskiy commented on src/components/Snackbar/Snackbar.tsx:
```diff
@@ -0,0 +1,95 @@
+    "non-closable": !closable,
+  });
+
+  let index = 1;
```
Если это Z-index то так и пиши

konkonskiy commented on src/components/Snackbar/Snackbar.tsx:
```diff
@@ -0,0 +1,95 @@
+    }
+    if (type === "warning") {
+      return Warning;
+    }
```
Ильдар, как такая конструкция называется?

konkonskiy commented on src/components/Snackbar/Snackbar.tsx:
```diff
@@ -0,0 +1,95 @@
+      );
+    }
+  }
+  return null;
```
Ну ты напридумал тут конечно) 
Избавляйся от таких условных конструкций

konkonskiy replied:
![image](https://user-images.githubusercontent.com/66328416/132170810-5e2cdd66-e7c4-47c8-b907-e17db40ed5d0.png)
Так лучше?

konkonskiy commented on src/Routes.tsx:
```diff
@@ -0,0 +1,43 @@
+    return authorizedRoutes;
+  } else {
+    return unauthorizedRoutes;
+  }
```
1. Такую конструкцию нужно заменять тренарником. 
2. Правильная логика, что ты используешь компоненту редирект. Но не правильно применил. Выводи все пути в свиче, это ошибка.

konkonskiy commented on src/Routes.tsx:
```diff
@@ -0,0 +1,43 @@
+    }
+  }, []);
+
+  const unauthorizedRoutes = (
```
Так делать не стоит

konkonskiy commented on src/store/User/UserActionCreators.ts:
```diff
@@ -0,0 +1,68 @@
+  };
+};
+
+export const exitUser = () => {
```
Проверял работу функции?

konkonskiy commented on src/pages/AutorizationPage/AuthorizationPage.tsx:
```diff
@@ -1,36 +1,74 @@
+    }
+  };
+
+  const linkClass = classNames("authorization__access", "link-default");
```
Зачем тут класснейм?

konkonskiy commented on src/store/Authorization/AuthorizationTypes.ts:
```diff
@@ -0,0 +1,26 @@
+  password: {
+    value: string;
+    isCorrect: boolean;
+  };
```
Вынеси в отдельный интерфейс, чтобы не дублировать логику

Feintty committed:
F-08 Переделаны роуты
F-08 Исправлен отлов ошибки в snackbar
F-08 Пофикшен логаут
F-08 Мелкие исправления

Feintty committed:
F-08 Убрана логика sncakbar'а из других компонентов

Feintty committed:
F-08 Рефакторинг в соответствии с замечанием

Feintty committed:
F-08 Фикс синтаксической ошибки

Feintty committed:
F-08 Фикс условной конструкции snackbar'а

FirsovRG commented on src/Routes.tsx:
```diff
@@ -0,0 +1,35 @@
+import { checkIsUserExists, exitUser } from "./store/User/UserActionCreators";
+
+const Routes = () => {
+  const { isLogged } = useTypedSelector((state) => state.user);
```
коллбэк селектора лучше выносить в отдельные функции, т.к. они могут использоваться в нескольких местах в проекте

FirsovRG commented on src/Routes.tsx:
```diff
@@ -0,0 +1,35 @@
+      </Route>
+      <Route path="/admin/orders">
+        <div>Вы уже авторизованы</div>
+        <button onClick={() => dispatch(exitUser())}>Выйти</button>
```
коллбэк лучше вынести в отдельную функцию

FirsovRG commented on src/components/Input/Input.tsx:
```diff
@@ -1,13 +1,34 @@
+    incorrect: !isCorrect,
+  });
+
+  const onInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
```
лучше не сокращать названия переменных даже для события. Название должно полностью отражать смысл переменной

FirsovRG replied:
и обработчики лучше называть со слова "handle"

FirsovRG commented on src/components/Snackbar/Snackbar.tsx:
```diff
@@ -0,0 +1,76 @@
+const Snackbar: React.FC<ISnackbarProps> = ({ snackbarId }) => {
+  const dispatch = useDispatch();
+  const { isOpened, id, type, message, closable, delay } = useTypedSelector(
+    (state) => state.snackbar
```
в отдельный селектор

FirsovRG commented on src/components/Snackbar/Snackbar.tsx:
```diff
@@ -0,0 +1,76 @@
+    "non-closable": !closable,
+  });
+
+  let zIndex = 1;
```
не совсем понятно, зачем здесь эта переменная, и почему она let

FirsovRG commented on src/components/Snackbar/Snackbar.tsx:
```diff
@@ -0,0 +1,76 @@
+    <div
+      onAnimationEnd={() => !closable && dispatch(snackbarClose(snackbarId))}
+      className={snackbarClassname}
+      style={{ animationDuration: `${delay}ms`, zIndex: zIndex++ }}
```
а через css нельзя определить этот z-index?

FirsovRG replied:
он вроде постоянный

FirsovRG commented on src/pages/AutorizationPage/AuthorizationPage.tsx:
```diff
@@ -1,36 +1,74 @@
+    dispatch(setPassword(""));
+  }, []);
+
+  const onLoginClick = (): void => {
```
вообще обработчики событий никогда ничего не возвращают, поэтому они всегда void. Не нужно здесь вручную типизировать

FirsovRG commented on src/store/Snackbar/SnackbarReducer.ts:
```diff
@@ -0,0 +1,35 @@
+      return {
+        ...initialSnackbar,
+        isOpened: true,
+        ...action.payload,
```
здесь неочевидно, что именно приходит в payload. Поэтому если объект небольшой - лучше расписывать по полям, либо весь payload записывать в одно типизированное поле

FirsovRG commented on src/store/User/UserActionCreators.ts:
```diff
@@ -0,0 +1,82 @@
+import errorCodeToMessage from "../../utils/errorCodeToMessage";
+
+export const login = (mail: string, password: string) => {
+  return async (dispatch: Dispatch<any>) => {
```
any

Feintty committed:
F-08 Фиксы в соответствии с замечаниями
